# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
#
# Filename: package/.../gcc/gcc-4.4.2-pure64-1.diff
# Copyright (C) 2009 - 2010 The OpenSDE Project
#
# More information can be found in the files COPYING and README.
#
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

diff -ruN gcc-4.4.2-orig/gcc/config/i386/linux64.h gcc-4.4.2/gcc/config/i386/linux64.h
--- gcc-4.4.2-orig/gcc/config/i386/linux64.h	2009-04-10 01:23:07.000000000 +0200
+++ gcc-4.4.2/gcc/config/i386/linux64.h	2009-12-31 11:20:50.000000000 +0100
@@ -58,8 +58,8 @@
    When the -shared link option is used a final link is not being
    done.  */
 
-#define GLIBC_DYNAMIC_LINKER32 "/lib/ld-linux.so.2"
-#define GLIBC_DYNAMIC_LINKER64 "/lib64/ld-linux-x86-64.so.2"
+#define GLIBC_DYNAMIC_LINKER32 "/lib32/ld-linux.so.2"
+#define GLIBC_DYNAMIC_LINKER64 "/lib/ld-linux-x86-64.so.2"
 
 #undef ASM_SPEC
 #define ASM_SPEC "%{v:-V} %{Qy:} %{!Qn:-Qy} %{n} %{T} %{Ym,*} %{Yd,*} \
diff -ruN gcc-4.4.2-orig/gcc/config/i386/t-linux64 gcc-4.4.2/gcc/config/i386/t-linux64
--- gcc-4.4.2-orig/gcc/config/i386/t-linux64	2008-07-02 17:59:19.000000000 +0200
+++ gcc-4.4.2/gcc/config/i386/t-linux64	2009-12-31 11:20:50.000000000 +0100
@@ -7,7 +7,7 @@
 
 MULTILIB_OPTIONS = m64/m32
 MULTILIB_DIRNAMES = 64 32 
-MULTILIB_OSDIRNAMES = ../lib64 $(if $(wildcard $(shell echo $(SYSTEM_HEADER_DIR))/../../usr/lib32),../lib32,../lib)
+MULTILIB_OSDIRNAMES = ../lib $(if $(wildcard $(shell echo $(SYSTEM_HEADER_DIR))/../../usr/lib32),../lib32,../lib32)
 
 LIBGCC = stmp-multilib
 INSTALL_LIBGCC = install-multilib
diff -ruN gcc-4.4.2-orig/gcc/config/mips/linux64.h gcc-4.4.2/gcc/config/mips/linux64.h
--- gcc-4.4.2-orig/gcc/config/mips/linux64.h	2009-02-20 16:20:38.000000000 +0100
+++ gcc-4.4.2/gcc/config/mips/linux64.h	2009-12-31 11:20:50.000000000 +0100
@@ -35,9 +35,9 @@
 %{!shared: \
   %{profile:-lc_p} %{!profile:-lc}}"
 
-#define GLIBC_DYNAMIC_LINKER32 "/lib/ld.so.1"
-#define GLIBC_DYNAMIC_LINKER64 "/lib64/ld.so.1"
-#define GLIBC_DYNAMIC_LINKERN32 "/lib32/ld.so.1"
+#define GLIBC_DYNAMIC_LINKER32 "/lib32/ld.so.1"
+#define GLIBC_DYNAMIC_LINKER64 "/lib/ld.so.1"
+#define GLIBC_DYNAMIC_LINKERN32 "/lib64/ld.so.1"
 #define UCLIBC_DYNAMIC_LINKERN32 "/lib32/ld-uClibc.so.0"
 #define LINUX_DYNAMIC_LINKERN32 \
   CHOOSE_DYNAMIC_LINKER (GLIBC_DYNAMIC_LINKERN32, UCLIBC_DYNAMIC_LINKERN32)
diff -ruN gcc-4.4.2-orig/gcc/config/mips/t-linux64 gcc-4.4.2/gcc/config/mips/t-linux64
--- gcc-4.4.2-orig/gcc/config/mips/t-linux64	2006-06-06 14:51:24.000000000 +0200
+++ gcc-4.4.2/gcc/config/mips/t-linux64	2009-12-31 11:20:50.000000000 +0100
@@ -1,6 +1,6 @@
 MULTILIB_OPTIONS = mabi=n32/mabi=32/mabi=64
 MULTILIB_DIRNAMES = n32 32 64
-MULTILIB_OSDIRNAMES = ../lib32 ../lib ../lib64
+MULTILIB_OSDIRNAMES = ../lib64 ../lib32 ../lib
 
 EXTRA_MULTILIB_PARTS=crtbegin.o crtend.o crtbeginS.o crtendS.o crtbeginT.o
 
diff -ruN gcc-4.4.2-orig/gcc/config/rs6000/linux64.h gcc-4.4.2/gcc/config/rs6000/linux64.h
--- gcc-4.4.2-orig/gcc/config/rs6000/linux64.h	2009-04-10 01:23:07.000000000 +0200
+++ gcc-4.4.2/gcc/config/rs6000/linux64.h	2009-12-31 11:20:50.000000000 +0100
@@ -342,8 +342,8 @@
 #undef	LINK_OS_DEFAULT_SPEC
 #define LINK_OS_DEFAULT_SPEC "%(link_os_linux)"
 
-#define GLIBC_DYNAMIC_LINKER32 "/lib/ld.so.1"
-#define GLIBC_DYNAMIC_LINKER64 "/lib64/ld64.so.1"
+#define GLIBC_DYNAMIC_LINKER32 "/lib32/ld.so.1"
+#define GLIBC_DYNAMIC_LINKER64 "/lib/ld64.so.1"
 #define UCLIBC_DYNAMIC_LINKER32 "/lib/ld-uClibc.so.0"
 #define UCLIBC_DYNAMIC_LINKER64 "/lib/ld64-uClibc.so.0"
 #if UCLIBC_DEFAULT
diff -ruN gcc-4.4.2-orig/gcc/config/rs6000/t-linux64 gcc-4.4.2/gcc/config/rs6000/t-linux64
--- gcc-4.4.2-orig/gcc/config/rs6000/t-linux64	2007-09-27 21:56:06.000000000 +0200
+++ gcc-4.4.2/gcc/config/rs6000/t-linux64	2009-12-31 11:20:50.000000000 +0100
@@ -19,7 +19,7 @@
 MULTILIB_EXTRA_OPTS     = fPIC mstrict-align
 MULTILIB_EXCEPTIONS     = m64/msoft-float
 MULTILIB_EXCLUSIONS     = m64/!m32/msoft-float
-MULTILIB_OSDIRNAMES	= ../lib64 $(if $(wildcard $(shell echo $(SYSTEM_HEADER_DIR))/../../usr/lib32),../lib32,../lib) nof
+MULTILIB_OSDIRNAMES	= ../lib $(if $(wildcard $(shell echo $(SYSTEM_HEADER_DIR))/../../usr/lib32),../lib32,../lib32) nof
 MULTILIB_MATCHES        = $(MULTILIB_MATCHES_FLOAT)
 
 softfp_wrap_start := '\#ifndef __powerpc64__'
diff -ruN gcc-4.4.2-orig/gcc/config/sparc/linux64.h gcc-4.4.2/gcc/config/sparc/linux64.h
--- gcc-4.4.2-orig/gcc/config/sparc/linux64.h	2009-02-20 16:20:38.000000000 +0100
+++ gcc-4.4.2/gcc/config/sparc/linux64.h	2009-12-31 11:22:47.000000000 +0100
@@ -109,8 +109,8 @@
 
 /* If ELF is the default format, we should not use /lib/elf.  */
 
-#define GLIBC_DYNAMIC_LINKER32 "/lib/ld-linux.so.2"
-#define GLIBC_DYNAMIC_LINKER64 "/lib64/ld-linux.so.2"
+#define GLIBC_DYNAMIC_LINKER32 "/lib32/ld-linux.so.2"
+#define GLIBC_DYNAMIC_LINKER64 "/lib/ld-linux.so.2"
 
 #ifdef SPARC_BI_ARCH
 
diff -ruN gcc-4.4.2-orig/gcc/config/sparc/t-linux64 gcc-4.4.2/gcc/config/sparc/t-linux64
--- gcc-4.4.2-orig/gcc/config/sparc/t-linux64	2006-02-08 08:19:43.000000000 +0100
+++ gcc-4.4.2/gcc/config/sparc/t-linux64	2009-12-31 11:24:07.000000000 +0100
@@ -1,6 +1,6 @@
 MULTILIB_OPTIONS = m64/m32
 MULTILIB_DIRNAMES = 64 32
-MULTILIB_OSDIRNAMES = ../lib64 ../lib
+MULTILIB_OSDIRNAMES = ../lib ../lib32
 
 LIBGCC = stmp-multilib
 INSTALL_LIBGCC = install-multilib
